### ISSUE TITLE
Consideration of Splice Junction Counts for Overlapping Paired-End Reads

### ISSUE THREAD

**Issue by cwarden45** [alexdobin/STAR#1712]
Hi,

In a [response to a preprint comment](http://disq.us/p/2rqfymh), it was mentioned that overlapping reads from smaller fragments are counted twice in the SJ.out.tab file created by STAR.

If this is true, is this something that the developers are aware of?  If so, is there a parameter that can currently be changed, or is that something planned for a future revision?

For example, by default, mpileup filters some reads, and the end result is that some strand-specific variants don't appear in the .pileup file.  That is for variant calling instead of splice junction analysis, but I think the idea of considering the overall fragment for a single splice junction count would be similar.

**I am not sure what is the absolute best test, but I defined overlapping sequences to cover a junction from GAPDH (*NM_001357943.2*):**

**Forward Read**:
```
@GAPDH/1
CCTGCACCACCAACTGCTTAGCACCCCTGGCCAAGGTCATCCATGACAACTTTGGTATCGTGGAAGGACTCATGACCACAGTCCATGCCATCACTGCCACC
+
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
```

**Reverse Read**:
```
@GAPDH/2
CGCGGCCATCACGCCACAGTTTCCCGGAGGGGCCATCCACAGTCTTCTGGGTGGCAGTGATGGCATGGACTGTGGTCATGAGTCCTTCCACGATACCAAAG
+
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
```

I then ran STAR with the following command:

```
/path/to/STAR --genomeDir $REF --readFilesIn $FQ1 $FQ2 --outFileNamePrefix $PREFIX --twopassMode Basic --outSAMstrandField intronMotif
```

Strictly speaking, I received an error message when using the exact sequences above.  I changed the End-Of-Line characters in Notepad++, but STAR kept thinking the quality score was one less than the sequence.  **So, I added an extra "B" in the quality scores for both forward and reverse reads to avoid that error message.**

In the resulting **GAPDH_test_SJ.out.tab** file, it looks like the junction was counted **correctly**:

```
chr12	6537391	6537583	1	1	1	1	0	27

```

Even without the 2-pass alignment, it looks like the junction was counted once:

```
chr12	6537391	6537583	1	1	0	1	0	27
```

Here is the view from IGV, so that each read really did cover the splice junction:

![GAPDH_test](https://user-images.githubusercontent.com/7663033/203446634-5324f321-83b1-466b-85c9-6a8ac8dbb90f.PNG)

**Is anybody familiar with encountering this issue in a different situation?**  I used *STAR v2.5* for the example above (with an alignment to hg38).

Thank you very much!

Sincerely,
Charles

**Comment by alexdobin** [comment]
Hi Charles,

the SJ.out.tab does *not* double count reads: if both mates of a read overlap the same junction, only one is added to the junction count.

Cheers
Alex

**Comment by cwarden45** [comment]
Hi Alex,

Thank you very much for confirming that the test example that I created should match most junctions.  In terms of my question as an individual, I will close this ticket.

I am not sure why something different was described in the preprint response.  However, I linked this ticket in my last response, and I hope this encourages discussion if other people have encountered issues in some other situation.

Thank you very much!

Sincerely,
Charles
